packages/eslint-plugin/src/rules/prefer-enum-initializers.ts

```diff
@@ -0,0 +1,42 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'prefer-enum-initializers',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Prefer initializing each enums member value',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    messages: {
+      defineInitializer:
+        "The value of the member '{{ name }}' should be explicitly defined",
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const sourceCode = context.getSourceCode();
+
+    return {
+      TSEnumDeclaration(node: TSESTree.TSEnumDeclaration): void {
+        const { members } = node;
+        const violatingMembers = members.filter(member => {
+          return member.initializer == null;
+        });
+
+        for (const member of violatingMembers) {
+          context.report({
+            node: member,
+            messageId: 'defineInitializer',
+            data: {
+              name: sourceCode.getText(member),
+            },
```

Comment:
Haven't done a full review yet, just musing based on a quick eyeball of the docs.

Do you think it's be worth adding "suggestion" fixers to this rule?
https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions

We could cover the two basic cases and give users a quick fix for them

For example, we could offer a few suggestion fixers:
1) add number initialiser equal to the 0-based index of the member
```ts
enum Foo {
  A,
  B, // can be suggestion fixed to B = 1,
}
```
2) add number initialiser equal to the 1-based index of the member
```ts
enum Foo {
  A,
  B, // can be suggestion fixed to B = 2,
}
```
3) add string initialiser equal to key name
```ts
enum Foo {
  Apple,
  Banana, // can be suggestion fixed to Banana = 'Banana',
}
```

Reply:
@bradzacher 

Hi, thank you for reviewing :bow:

I agree with you.
Let me try to add that feature.